Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/ui/StringProvider.kt

```diff
@@ -0,0 +1,14 @@
+package de.rki.coronawarnapp.util.ui
+
+import android.content.Context
+import androidx.annotation.StringRes
+
+interface StringProvider {
+    fun getString(@StringRes resId: Int, formatArg: Any? = null): String
+}
+
+class ContextStringProvider(private val context: Context) : StringProvider {
+    override fun getString(@StringRes resId: Int, formatArg: Any?): String {
```

Comment:
To make it flexible for other cases  `varargs formatArg: Any?` as in `context.getString`

Reply:
Our current lint setup does not allow varargs.

Reply:
Would be okay here to use `Suppress`, the performance impact of `varargs` shouldn't be an issue in this case. Maybe we should relax the LINT check for this, this is rarely  problematic and Kotlin `varargs` is really nice :smiley: .